givitsite/friendreq/tests.py

```diff
@@ -1,3 +1,47 @@
-from django.test import TestCase
+from django.test import TestCase 
+from django.shortcuts import reverse
+from django.contrib.auth.models import User
+from .models import ItemRequest,ItemsFound
+from django.conf import settings
 
-# Create your tests here.
+class FriendPageTests(TestCase):
+    # check the status code when navigate to request page
+    def test_request_status_code(self):
+        response = self.client.get('/friend/request')
+        self.assertEquals(response.status_code,200)
+
+    # check url name
+    def test_request_url_name(self):
+        response = self.client.get(reverse('itemRequest_create_view'))
+        self.assertEquals(response.status_code,200)
+
+    # check the status code when navigate to feed page
+    def test_feed_status_code(self):
+        response = self.client.get('/friend/feed')
+        self.assertEquals(response.status_code,200)
+
+    # check url name
+    def test_feed_url_name(self):
+        response = self.client.get(reverse('requestItem'))
+        self.assertEquals(response.status_code,200)
+
+    # check if the correct template is being rendered
+    def test_feed_correct_tamplate(self):
+        response = self.client.get(reverse('requestItem'))
+        self.assertEquals(response.status_code,200)
+        self.assertTemplateUsed(response,'feed.html')
+
+class ItemRequest_test(TestCase):
+    @classmethod
+    def setUp(cls):
+        test_user = User.objects.filter(is_superuser=True)
+        ItemRequest.objects.create(special_req = "this is a Test", friend_id =test_user[0])
+    
+    def test_text(self):
```

Comment:
I'm not sure what this is testing exactly, I'm only seeing calls to ORM functions and nothing calls any code that you wrote... You're supposed to test your own code not the ORM....

Reply:
understood:
I'll extend it to check the views (not just the  model)
I'm still having a problem with setting new non-superuser since (wait for a specific merge).